Formula/fastnetmon.rb

```diff
@@ -0,0 +1,44 @@
+class Fastnetmon < Formula
+  desc "DDoS detection tool with sFlow, Netflow, IPFIX and port mirror support"
+  homepage "https://github.com/pavel-odintsov/fastnetmon/"
+  url "https://github.com/pavel-odintsov/fastnetmon/archive/14481a07685171de11a6be61895d83a3dc51d763.tar.gz"
+  version "1.2.2"
+  sha256 "c581a9bcdf416b1b585ffa9ca7de6dcd1a5778d3da6bc6c948fdd87009686a75"
+  license "GPL-2.0-only"
+
+  depends_on "cmake" => :build
+  depends_on "boost"
+  depends_on "capnp"
+  depends_on "grpc"
+  depends_on "hiredis"
+  depends_on "json-c"
+  depends_on "log4cpp"
+  depends_on "mongo-c-driver"
+  depends_on "openssl@1.1"
+
+  def install
+    system "cmake", "-S", "src", "-B", "build",
+      "-DENABLE_CUSTOM_BOOST_BUILD=FALSE",
+      "-DDO_NOT_USE_SYSTEM_LIBRARIES_FOR_BUILD=FALSE",
+      "-DCMAKE_SYSTEM_INCLUDE_PATH=/usr/local/opt/openssl\@1/include",
+      "-DCMAKE_SYSTEM_LIBRARY_PATH=/usr/local/opt/openssl\@1/lib", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The test does not look, can you check the other formulae tests to get some ideas?

Reply:
Thank you for feedback. I reworked test completely. Instead of simple configuration check I do full run, wait for few iterations of traffic processing and then check that it properly generates traffic reports (files) for IPv4 and IPv6. So it tests main feature of our tool using all steps.